### PR TITLE
Add payment cards endpoints to SDK

### PR DESCRIPTION
**Pull request by tylerwoonton**
We need to add the payment cards endpoints to the SDK.

The following endpoints are added by this PR:

GET `/cards`
GET `/cards/{id}`
POST `/cards`
PATCH `/cards/{id}`
DELETE `/cards/{id}`

### CODE REVIEW

File: src/Billing/PaymentCardClient.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /**
+     * @param $paymentCard
+     * @return SelfResponse
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function create($paymentCard)
+    {
+        $response = $this->post("v1/cards", $this->friendlyToApi($paymentCard, self::MAP));
```
Comment (Gman98ish) [inline src/Billing/PaymentCardClient.php]: This need to be json encoded e.g.

```php
$response = $this->post("v1/cards", json_encode($this->friendlyToApi($paymentCard, self::MAP)));
```

File: src/Billing/PaymentCardClient.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    /**
+     * @param $id
+     * @param $paymentCard
+     * @return SelfResponse
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function update($id, $paymentCard)
+    {
+        $response = $this->patch("v1/cards/$id", $this->friendlyToApi($paymentCard, self::MAP));
```
Comment (Gman98ish) [inline src/Billing/PaymentCardClient.php]: Same here

**Comment by tylerwoonton** [comment]
@Gman98ish I've resolved your suggestions - could you approve if you're happy? 👍

**Comment by Gman98ish** [comment]
You need to fix the tests and merge master first @tylerwoonton

**Comment by tylerwoonton** [comment]
@Gman98ish Sorry about that, fixed now.

File: src/Billing/PaymentCardClient.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    /**
+     * @param $item
+     * @return PaymentCard
+     */
+    protected function serializePaymentCard($item)
+    {
+        $paymentCard = new PaymentCard($this->apiToFriendly($item, self::MAP));
+
+        return $paymentCard;
```
Comment (MoHangingWithLegends) [inline src/Billing/PaymentCardClient.php]: just a suggestion for next time can just `return new PaymentCard($this->apiToFriendly($item, self::MAP));`

File: tests/Billing/PaymentCardClientTest.php [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        $this->assertEquals("M44 0AB", $paymentCard->postcode);
+        $this->assertEquals("4111111111111111", $paymentCard->cardNumber);
+        $this->assertEquals("Visa", $paymentCard->cardType);
+        $this->assertEquals('2019-10-31T00:00:00+0000', $paymentCard->validFrom->format(DateTime::ISO8601));
+        $this->assertEquals('2035-10-31T00:00:00+0000', $paymentCard->expiry->format(DateTime::ISO8601));
+        $this->assertEquals(12, $paymentCard->issueNumber);
+        $this->assertTrue(true, $paymentCard->primaryCard);
+
+        $this->assertInternalType('int', $paymentCard->id);
+        $this->assertInternalType('int', $paymentCard->issueNumber);
```
Comment (MoHangingWithLegends) [inline tests/Billing/PaymentCardClientTest.php]: nice 👍

File: src/Billing/PaymentCardClient.php (hunk 3 above)
Comment (tylerwoonton) [reply src/Billing/PaymentCardClient.php]: Yeah. I just preferred the separation. 

Comes down to personal preference in the end but thanks for raising it! 👍

File: src/Billing/Entities/PaymentCard.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * @property int $issueNumber
+ * @property boolean $primaryCard
+ */
+class PaymentCard extends Entity
+{
+    protected $dates = ['validFrom', 'expiry'];
+
+    public function isExpired()
+    {
+        return $this->expiry < date_format(new DateTime, 'Y-m-d');
```
[the remark this answers is not in the document]
Comment (tylerwoonton) [reply src/Billing/Entities/PaymentCard.php]: I've tested that this still produces the intended result. I'm happy to apply the suggestion.

**Pull request merged**